### PR TITLE
Lsp clojure

### PR DESCRIPTION
**Pull request by RohanVashisht1234**
I have added lsp clojure

### CODE REVIEW

File: manifest.json [hunk 1]
```diff
@@ -23,6 +23,53 @@
     ],
     "dependencies": { "language_c": {}, "lsp": {} }
   },{
+  "addons": [{
+    "id": "lsp_clojure",
+    "description": "Automatic configuration/binary download for LSP completion for clojure with clojure-lsp.",
+    "path": "plugins/lsp_clojure.lua",
+    "version": "2024.03.31-19.10.13",
```
Comment (Guldoman) [inline manifest.json:30]: We don't support dashes in the version.
It's probably fine to just drop the dash and anything after it.

File: manifest.json [hunk 2]
```diff
@@ -23,6 +23,53 @@
...
+        "url": "https://github.com/clojure-lsp/clojure-lsp/releases/download/2024.03.31-19.10.13/clojure-lsp-native-windows-amd64.zip",
+        "arch": "x86_64-windows",
+        "checksum": "c29acff1dc06f5810b055c67722e7109bcc8c80b80f24feac37998a8320930e0"
+      },
+      {
+        "url": "https://github.com/clojure-lsp/clojure-lsp/releases/download/2024.03.31-19.10.13/clojure-lsp-native-windows-amd64.zip",
+        "arch": "amd64-windows",
+        "checksum": "c29acff1dc06f5810b055c67722e7109bcc8c80b80f24feac37998a8320930e0"
+      }
+    ],
```
Comment (Guldoman) [inline manifest.json:67]:
```suggestion
    "files": [
      {
        "url": "https://github.com/clojure-lsp/clojure-lsp/releases/download/2024.03.31-19.10.13/clojure-lsp-native-linux-aarch64.zip",
        "arch": "aarch64-linux",
        "checksum": "0715435335d09f03deebbe946275196cac400f3b379dfb0bd55a060fe46d49b4"
      },
      {
        "url": "https://github.com/clojure-lsp/clojure-lsp/releases/download/2024.03.31-19.10.13/clojure-lsp-native-linux-amd64.zip",
        "arch": "x86_64-linux",
        "checksum": "3d860d25e7c354198fa9c975ba01bfb27d1aba42d8e334f0e529bbfdc5c1a5e6"
      },
      {
        "url": "https://github.com/clojure-lsp/clojure-lsp/releases/download/2024.03.31-19.10.13/clojure-lsp-native-macos-aarch64.zip",
        "arch": "aarch64-darwin",
        "checksum": "78ada7b23d604f1542efea29e283baa26f3e9011450929d876827cc148817f62"
      },
      {
        "url": "https://github.com/clojure-lsp/clojure-lsp/releases/download/2024.03.31-19.10.13/clojure-lsp-native-macos-amd64.zip",
        "arch": "x86_64-darwin",
        "checksum": "ec8468a2aef17ee365c0d4f79a069f441c5fe9d466286929af012060dbf68a35"
      },
      {
        "url": "https://github.com/clojure-lsp/clojure-lsp/releases/download/2024.03.31-19.10.13/clojure-lsp-native-windows-amd64.zip",
        "arch": "x86_64-windows",
        "checksum": "c29acff1dc06f5810b055c67722e7109bcc8c80b80f24feac37998a8320930e0"
      }
    ],
```

File: plugins/lsp_clojure.lua [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  if ARCH == "aarch64-linux" then
+    filename = "clojure-lsp"
+  else
+    filename = "clojure-lsp"
+  end
+end
+
+lspconfig.rust_analyzer.setup(common.merge({
+  command = { installed_path .. PATHSEP .. filename }
+}, config.plugins.lsp_rust or {}))
```
Comment (Guldoman) [inline plugins/lsp_clojure.lua:29]:
```suggestion
lspconfig.clojure_lsp.setup(common.merge({
  command = { installed_path .. PATHSEP .. filename }
}, config.plugins.lsp_clojure or {}))
```

File: plugins/lsp_clojure.lua [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  else
+    filename = "clojure-lsp"
+  end
+else
+  if ARCH == "aarch64-linux" then
+    filename = "clojure-lsp"
+  else
+    filename = "clojure-lsp"
+  end
+end
```
Comment (Guldoman) [inline plugins/lsp_clojure.lua:25]:
```suggestion
local filename = "clojure-lsp" .. (PLATFORM == "Windows" and ".exe" or "")
```

**Review by Guldoman — CHANGES_REQUESTED** [review]
Also format the manifest please.

**Comment by RohanVashisht1234** [comment]
I think I have resolved all the issues

**Comment by RohanVashisht1234** [comment]
yes, also I have changed the linux-native to linux-static-native for better compatibility

**Comment by RohanVashisht1234** [comment]
ok, everything has been fixed, added a few things also

File: plugins/lsp_clojure.lua [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+-- mod-version:3
+
+local lspconfig = require "plugins.lsp.config"
+local common = require "core.common"
+local config = require "core.config"
+
+local installed_path = USERDIR .. PATHSEP .. "plugins" .. PATHSEP .. "lsp_clojure"
+
+lspconfig.clojure_lsp.setup(common.merge({
+  command = { installed_path .. PATHSEP .. filename }
```
Comment (Guldoman) [inline plugins/lsp_clojure.lua:12]: You aren't defining `filename`.

**Review by Guldoman — CHANGES_REQUESTED** [review]
It looks like you're changing a lot of unrelated entries in the manifest. Could you revert that?

**Comment by RohanVashisht1234** [comment]
ok, fixed that also

**Comment by RohanVashisht1234** [comment]
Yes, changed the formatting of the json to origionals

**Review by Guldoman — APPROVED** [review]
Looks good!

Tested on Linux and it seems to work fine without any additional dependency.
Can anybody test on Windows and macOS?

**Comment by RohanVashisht1234** [comment]
Has been tested to be working perfectly well on macos  aarch 64

https://media.discordapp.net/attachments/847122429742809211/1228412928034734090/rohanvashishtmactesting.mov?ex=662bf3b0&is=66197eb0&hm=d9e966c109fe062089e0e7fed35509015d956e01c5f5ea8132a45fcfa0ee6492&

**Comment by Guldoman** [comment]
Seems to work fine on Windows too.
Waiting on lite-xl/lite-xl-plugins#421 to be merged.

**Pull request merged**